src/test/java/com/powsybl/openloadflow/sa/WoodburyDcSecurityAnalysisWithActionsTest.java

```diff
@@ -0,0 +1,427 @@
+        List<Action> actions = List.of(new PhaseTapChangerTapPositionAction("pstAbsChange", "PS1", false, 0),
+                new PhaseTapChangerTapPositionAction("pstRelChange", "PS1", true, -1));
+        List<OperatorStrategy> operatorStrategies = List.of(
+                new OperatorStrategy("strategyTapAbsChange", ContingencyContext.specificContingency("L1"), new TrueCondition(), List.of("pstAbsChange")),
+                new OperatorStrategy("strategyTapRelChange", ContingencyContext.specificContingency("L1"), new TrueCondition(), List.of("pstRelChange")));
+
+        SecurityAnalysisResult result = runSecurityAnalysis(network, contingencies, monitors, securityAnalysisParameters,
+                operatorStrategies, actions, ReportNode.NO_OP);
+
+        assertNotNull(result);
+
+        OperatorStrategyResult resultAbs = getOperatorStrategyResult(result, "strategyTapAbsChange");
+        BranchResult brAbsL2 = resultAbs.getNetworkResult().getBranchResult("L2");
+        BranchResult brAbsPS1 = resultAbs.getNetworkResult().getBranchResult("PS1");
+
+        OperatorStrategyResult resultRel = getOperatorStrategyResult(result, "strategyTapRelChange");
+        BranchResult brRelL2 = resultRel.getNetworkResult().getBranchResult("L2");
+        BranchResult brRelPS1 = resultRel.getNetworkResult().getBranchResult("PS1");
+
+        // Apply contingency by hand
+        network.getLine("L1").getTerminal1().disconnect();
+        network.getLine("L1").getTerminal2().disconnect();
+        // Apply remedial action
+        network.getTwoWindingsTransformer("PS1").getPhaseTapChanger().setTapPosition(0);
+        OpenLoadFlowParameters.create(parameters);
+
+        loadFlowRunner.run(network, parameters);
+
+        // Compare results on the line L2
+        assertEquals(network.getLine("L2").getTerminal1().getP(), brAbsL2.getP1(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(network.getLine("L2").getTerminal2().getP(), brAbsL2.getP2(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(network.getLine("L2").getTerminal1().getP(), brRelL2.getP1(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(network.getLine("L2").getTerminal2().getP(), brRelL2.getP2(), LoadFlowAssert.DELTA_POWER);
+        // Compare results on the t2wt PS1
+        assertEquals(network.getTwoWindingsTransformer("PS1").getTerminal1().getP(), brAbsPS1.getP1(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(network.getTwoWindingsTransformer("PS1").getTerminal2().getP(), brAbsPS1.getP2(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(network.getTwoWindingsTransformer("PS1").getTerminal1().getP(), brRelPS1.getP1(), LoadFlowAssert.DELTA_POWER);
+        assertEquals(network.getTwoWindingsTransformer("PS1").getTerminal2().getP(), brRelPS1.getP2(), LoadFlowAssert.DELTA_POWER);
+    }
+
+    @Test
+    void testSaDcPhaseTapChangerTapPositionChangeWithConnectivityBreak() {
+        Network network = PhaseControlFactory.createNetworkWith3Buses();
```

Comment:
This test doesn't really test a connectivity break. The remaining part after the L23 contingency is just an isolated bus without any branches.

Reply:
Thanks for the clarification Pierre. I understand that this test is more about what happens to the main component when many injections are lost due to a split of component in two parts, than what happens to the separate component.
And it is a very useful test since it shows this does not work well in default DC mode.